### PR TITLE
Nsfw as mask

### PR DESCRIPTION
**Pull request by isabella618033**

### CODE REVIEW

**Review by opentaco — APPROVED** [review]

**Review by Eugene-hu — APPROVED** [review]

File: openvalidators/forward.py [hunk 1]
```diff
@@ -73,10 +73,18 @@ async def run_step( self, prompt: str, k: int, timeout: float, name: str, exclud
     # Compute the rewards for the responses gien the prompt.
     rewards:torch.FloatTensor = torch.ones( len( responses ), dtype=torch.float32).to(self.device)
```
Comment (steffencruz) [inline openvalidators/forward.py:74]: If rewards are summed over the models list, shouldn't this be `torch.zeros`?

File: openvalidators/reward/blacklist.py [hunk 2]
```diff
@@ -29,9 +29,16 @@ class Blacklist( BaseRewardModel ):
     def name(self) -> str: return "blacklist"
 
     def reward( self, prompt: str, completion: str, name: str ) -> float:
-        if completion in blacklist: return 0.0
-        else: return 1
+        if completion in blacklist: 
+            return 0.0
+        
+        if completion == prompt:
```
Comment (steffencruz) [inline openvalidators/reward/blacklist.py:35]: I'm not keen on this form of blacklist. It's very rigid and brittle, and is just doing the job of a relevance+reward model. A few improvements would be

- Regex matching
- [Fuzzy](https://pypi.org/project/fuzzywuzzy/) matching

If we are going to continue as-is, we can at least improve the coverage by replacing strictly equals with substring matching.
```python
if any(b in completion for b in blacklist):
    ...
```

File: openvalidators/reward/relevance.py [hunk 3]
```diff
@@ -92,7 +92,10 @@ def reward( self, prompt: str, completion:str , name: str ) -> float:
         diff = (( completion_embedding - prompt_embedding )**2).mean()**0.5
 
         # Return relevance scoring.
-        return float(-diff)
+        return 0.0 if float(-diff) < bound else 1.0
 
     def get_rewards( self, prompt: str, completions: List[str], name: str ) -> torch.FloatTensor:
-        return torch.tensor( [self.reward( prompt, completion, name ) for completion in completions], dtype=torch.float32).to(self.device)
\ No newline at end of file
+        return torch.tensor( [self.reward( prompt, completion, name ) for completion in completions], dtype=torch.float32).to(self.device)
```
Comment (steffencruz) [inline openvalidators/reward/relevance.py:98]: This seems completely general. Why not push this into the base class?

File: openvalidators/reward/relevance.py [hunk 4]
```diff
@@ -92,7 +92,10 @@ def reward( self, prompt: str, completion:str , name: str ) -> float:
...
+    
+    def normalize_rewards( self, rewards: torch.FloatTensor ) -> torch.FloatTensor:
```
Comment (steffencruz) [inline openvalidators/reward/relevance.py:100]: Also seems general..

File: openvalidators/reward/nsfw.py [hunk 5]
```diff
@@ -37,26 +37,30 @@ def __init__( self, device: str ):
 
     def reward( self, prompt: str, completion: str, name: str ) -> float:
```
Comment (steffencruz) [inline openvalidators/reward/nsfw.py:38]: We should use a consistent API for all reward methods. Right the the design is inconsistent, two options are:

1. Passing bound as an optional argument in all `reward` methods
2. Defining bound in config (since the user doesn't change the number during runtime anyway)
